### DISCUSSION TITLE
Search front matter

### DISCUSSION THREAD

**Discussion by aalexei** [protesilaos/denote#338] [Ideas]
Hi, is there a nice tool in Emac's ecosystem that would allow me to search the frontmatter (nicely presenting the matches in consult say) but resolve to the actual filename?

The problem I have in mind are reference notes, where each note to a published paper is titled by it's citekey. Something like "2022ger367.org" which I use all over my system.  I want to be able to search based on more meaningful fields like authors, full title, or full reference. All of those fields can be added to frontmatter keywords.

**Comment by protesilaos** [comment]
> From: Alexei ***@***.***>
> Date: Mon,  6 May 2024 20:50:51 -0700
>
> Hi, is there a nice tool in Emac's ecosystem that would allow me to
> search the frontmatter (nicely presenting the matches in consult say)
> but resolve to the actual filename?

Can this not be done with 'consult-grep' or 'consult-ripgrep'?

Note that you need to be in the 'denote-directory' for this to work.

[ I am developing consult-denote to add a grep from anywhere, among
others: https://github.com/protesilaos/consult-denote ]

-- 
Protesilaos Stavrou
https://protesilaos.com

**Reply by aalexei** [reply]
Thanks for that. That get's me some of the way towards the workflow I want. After some digging, the citar package and citar-denote integration might be what I'm after, or something I can adapt.

**Comment by protesilaos** [comment]
> From: Alexei ***@***.***>
> Date: Sat, 11 May 2024 20:04:15 -0700
>
> Thanks for that. That get's me some of the way towards the workflow I
> want. After some digging, the citar package and citar-denote
> integration might be what I'm after, or something I can adapt.

Sounds good!

-- 
Protesilaos Stavrou
https://protesilaos.com
